### PR TITLE
config: handle libinput <device category=""> as default

### PR DESCRIPTION
**Pull request by johanmalm**
Fixes: #1382

### CODE REVIEW

File: src/config/libinput.c [hunk 1]
```diff
@@ -26,7 +26,7 @@ libinput_category_init(struct libinput_category *l)
 enum device_type
 get_device_type(const char *s)
 {
-	if (!s) {
+	if (!s || !*s) {
 		return DEFAULT_DEVICE;
 	}
```
Comment (Consolatis) [inline src/config/libinput.c:31]: This is technically not needed, none of the following would match and we'd return `DEFAULT_DEVICE` anyway.

**Comment by Consolatis** [comment]
LGTM

File: src/seat.c [hunk 2]
```diff
@@ -57,6 +57,37 @@ device_type_from_wlr_device(struct wlr_input_device *wlr_input_device)
...
 
+/*
+ * Get applicable profile (category) by matching first by name and secondly be
+ * type (e.g. 'touch' and 'non-touch'). If not suitable match is found based on
+ * those two criteria we fallback on 'default'.
+ */
+static struct libinput_category *
+get_category(struct wlr_input_device *device)
+{
+	enum device_type type = device_type_from_wlr_device(device);
```
Comment (domo141) [inline src/seat.c]: it looks to me that if this could be done in line 79 (split in 2 if c89 required)

File: src/seat.c [hunk 3]
```diff
@@ -57,6 +57,37 @@ device_type_from_wlr_device(struct wlr_input_device *wlr_input_device)
...
+
+	/* By name */
+	struct libinput_category *category = NULL;
```
Comment (domo141) [inline src/seat.c:71]: does compiler optimize the writing of NULL to category away.

I did 

`git grep -B1 wl_list_for_each | less`

and did not see settings such vars set to NULL (in the lines it printed out)

File: include/config/libinput.h [hunk 4]
```diff
@@ -7,6 +7,9 @@
 #include <wayland-server-core.h>
 
 enum device_type {
+	LAB_LIBINPUT_DEVICE_NONE = 0,
+
+	// FIXME: Rename the entries below with a LAB_LIBINPUT_ prefix
```
Comment (domo141) [inline include/config/libinput.h]: comment could be ... LAB_LIBINPUT_DEVICE_ prefix ...

**Review by domo141 — APPROVED** [review]
LGTM AFAIU, with just some thoughts in comments...

File: src/seat.c (hunk 2 above)
Comment (johanmalm) [reply src/seat.c]: Good point. Have moved it.

File: src/seat.c (hunk 3 above)
Comment (johanmalm) [reply src/seat.c]: Fair point. Have removed =NULL

File: include/config/libinput.h (hunk 4 above)
Comment (johanmalm) [reply include/config/libinput.h]: Overtaken by events. Have already sorted out the enums.

**Pull request merged**